docs/source/upgrade.rst

```diff
@@ -10,6 +10,36 @@ Most changes are carefully designed not to break existing code; however changes
 Chainer v5
 ==========
 
+FunctionNode Classes are Hidden from ``chainer.functions``
+----------------------------------------------------------
+
+Prior to Chainer v5, :class:`~chainer.FunctionNode` classes (e.g., ``chainer.functions.MaxPooling2D``) are exposed under :mod:`chainer.functions`.
+In Chainer v5, these classes are hidden from :mod:`chainer.functions`.
+Use the equivalent wrapper functions listed in :doc:`reference/functions` (e.g., :func:`chainer.functions.max_pooling_2d`) instead.
+
+Some wrapper functions now provide options to return internal states to avoid directly using :class:`~chainer.FunctionNode` classes.
+
+* :func:`chainer.functions.max_pooling_2d`: ``return_indices``
+* :func:`chainer.functions.max_pooling_nd`: ``return_indices``
+* :func:`chainer.functions.dropout`: ``mask``, ``return_mask``
+* :func:`chainer.functions.gaussian`: ``eps``, ``return_eps``
+
+For example, suppose your existing code needs to access ``MaxPooling2D.indexes`` to later perform upsampling::
+
+    p = F.MaxPooling2D(2, 2)
+    h = p.apply((x,))[0]
+    ...
+    F.upsampling_2d(h, p.indexes, ...)
```

Comment:
`...` in the arguments is not there in the v5 example.  Could you remove it?

Reply:
Fixed.